.travis.yml

```diff
@@ -3,26 +3,28 @@
 sudo: false
 cache:
   directories:
-    - $HOME/download
-    - $HOME/download/miniconda.sh
     - $HOME/.cache/pip
     - $HOME/.theano
+    - $HOME/download # Sufficient to add miniconda.sh to TRAVIS cache.
+    - $HOME/miniconda2 # Add the installation to TRAVIS cache.
 
 
 language: python
 python:
   - "2.7"
   - "3.3"
 
+# NB:
+# In before_install and install sections below,
+# some codes have been moved to separate files
+# to better handle if-else shell syntax
+# for multiple lines. New files are in
+# new folder ".travis".
+
 # command to install dependencies
 before_install:
-# Install miniconda to avoid compiling scipy
-  - mkdir -p download
-  - cd download
-  - wget -c https://repo.continuum.io/miniconda/Miniconda2-4.1.11-Linux-x86_64.sh -O miniconda.sh
-  - chmod +x miniconda.sh
-  - ./miniconda.sh -b
-  - cd ..
+  - chmod +x ./.travis/travis_before_install.sh
+  - ./.travis/travis_before_install.sh
```

Comment:
I think you can mark the file as executable from git itself, and you would not need to chmod it each time.